crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -920,10 +919,10 @@ impl<'db> TypeInferenceBuilder<'db> {
         let ast::ExprNumberLiteral { range: _, value } = literal;
 
         match value {
-            ast::Number::Int(n) => {
-                // TODO support big int literals
```

Comment:
Does that mean that type narrowing with big int literals won't be supported because they always infer to `int`?

Should we use the parsers `Int` type?

Reply:
I think we discussed this when I added the `IntLiteral` type. We probably can use the parser's `Int` type to support big integers. I think I even tried doing that, but ran into some complexities (I just don't remember what they were now.)

This only impacts integer literals outside the i64 range. In my experience these are extremely uncommon (2^63 is a really large number, as you can see in the massive integer literal I added in the test here.) So I'm not sure it's really worth adding complexity or cost to the implementation to support them. Tracking exact literal values of integer types is a best-effort feature anyway; there are many cases where we'll just have to fallback to "it's an int." But we can see if it causes anyone problems in practice.